File: packages/csharp/LiterLlm/ProviderConfig.cs
namespace LiterLlm;

/// <summary>
/// Configuration for a custom LLM provider not in the built-in registry.
/// </summary>
/// <param name="Name">Unique provider name used for model routing.</param>
/// <param name="BaseUrl">The provider's API base URL.</param>
/// <param name="AuthHeader">The header name used for authentication (e.g. "Authorization").</param>
/// <param name="ModelPrefixes">Model name prefixes that route to this provider.</param>
public record ProviderConfig(
    string Name,
    string BaseUrl,
    string AuthHeader,
    IReadOnlyList<string> ModelPrefixes);

File: packages/java/src/main/java/dev/kreuzberg/literllm/ProviderConfig.java
package dev.kreuzberg.literllm;

import java.util.List;

/**
 * Configuration for a custom LLM provider not in the built-in registry.
 *
 * @param name
 *            unique provider name used for model routing
 * @param baseUrl
 *            the provider's API base URL
 * @param authHeader
 *            the header name used for authentication (e.g. "Authorization")
 * @param modelPrefixes
 *            model name prefixes that route to this provider
 */
public record ProviderConfig(String name, String baseUrl, String authHeader, List<String> modelPrefixes) {
}
